sympy/physics/mechanics/joint.py

```diff
@@ -518,23 +703,25 @@ def _generate_speeds(self, speed):
         return speeds
 
     def _orient_frames(self):
-        frame = self._set_orientation()
-        self.child.frame.orient_axis(frame, self.parent_axis,
-                                    self.coordinates[0])
+        self._joint_axis = self._axis(
+            self.joint_axis, self.parent_interframe, self.child_interframe)
+        axis = self._express_joint_axis(self.parent_interframe)
+        self.child_interframe.orient_axis(
+            self.parent_interframe, axis, self.coordinates[0])
 
     def _set_angular_velocity(self):
-        self.child.frame.set_ang_vel(self.parent.frame, self.speeds[0] *
-                                     self.parent_axis.normalize())
+        self.child_interframe.set_ang_vel(self.parent_interframe, self.speeds[
+            0] * self.joint_axis.normalize())
 
     def _set_linear_velocity(self):
```

Comment:
Currently the velocities are only set in the body frames not also in the intermediate frames. This means that these intermediate frames can still be translating with respect to their body, which should not be the case.
Either some extra `set_vel`'s should be added or `self.parent_point.set_vel(self.parent.frame, 0)` should be changed to `self.parent_point.set_vel(self.parent_interframe, 0)` (same for the child).

Which option would be best?

Reply:
Here is a simple experiment:
```py
from sympy import pi
from sympy.physics.mechanics import Body, PrismaticJoint, ReferenceFrame

P, C = Body('P'), Body('C')
Pint, Cint = ReferenceFrame('Pint'), ReferenceFrame('Cint')
Pint.orient_axis(P.frame, P.frame.x, pi)
Cint.orient_axis(C.frame, C.frame.y, -pi / 2)
joint = PrismaticJoint('P', P, C, parent_interframe=Pint, child_interframe=Cint,
                       parent_point=P.x, child_point=-C.y, joint_axis=P.z)
P.masscenter.vel(joint.parent_interframe)
ValueError: Velocity of point P_masscenter has not been defined in ReferenceFrame Pint
```

Reply:
We could also just add a `body.masscenter.set_vel(int_frame, 0)` in the interframe creation method.